static/app/components/core/compactSelect/compactSelect.tsx

```diff
@@ -22,6 +23,11 @@ export type {SelectOption, SelectOptionOrSection, SelectSection, SelectKey};
 interface BaseSelectProps<Value extends SelectKey>
   extends Omit<ControlProps, 'onClear' | 'clearable'> {
   options: Array<SelectOptionOrSection<Value>>;
+  /**
+   * Number of options above which virtualization will be enabled.
+   * @default 100
+   */
+  virtualThreshold?: number;
```

Comment:
I'm having second thoughts about exposing an arbitrary override. On one hand, it's good API practice, but on the other, we can be almost certain that after 150 items, the performance will be impacted, and this will become a problem.

Wdyt about about enforcing an internal max of 150, and only allowing users to decrease the threshold? 

```suggestion
  virtualizeThreshold?: number;
```